homeassistant/components/lutron/__init__.py

```diff
@@ -64,19 +81,25 @@ def setup(hass: HomeAssistant, base_config: ConfigType) -> bool:
         "scene": [],
         "binary_sensor": [],
     }
+    host = config_entry.data[CONF_HOST]
+    uid = config_entry.data[CONF_USERNAME]
+    pwd = config_entry.data[CONF_PASSWORD]
 
-    config = base_config[DOMAIN]
-    hass.data[LUTRON_CONTROLLER] = Lutron(
-        config[CONF_HOST], config[CONF_USERNAME], config[CONF_PASSWORD]
-    )
+    def _connect() -> bool:
+        hass.data[LUTRON_CONTROLLER].load_xml_db()
+        return True
 
-    hass.data[LUTRON_CONTROLLER].load_xml_db()
+    hass.data[LUTRON_CONTROLLER] = Lutron(host, uid, pwd)
+    await hass.async_add_executor_job(_connect)
```

Comment:
```suggestion
    await hass.async_add_executor_job(hass.data[LUTRON_CONTROLLER].load_xml_db)
```
I just read that comment I left, and I was trying to figure out what I was trying to say but I couldn't figure out, I think it was this.

Reply:
that makes sense.  change made.

Reply:
I reverted this as it throws errors otherwise.  when putting the method directly in the executor instead of in the _connect it throws: RuntimeError: Blocking calls must be done in the executor or a separate thread; Use `await hass.async_add_executor_job()`; 
I had forgotten I had already been down that road.  I did change it from connect to load db to be more clear